core/spec/models/spree/order_cancellations_spec.rb

```diff
@@ -1,6 +1,64 @@
 require 'spec_helper'
 
 describe Spree::OrderCancellations do
+  describe "#cancel_unit" do
+    subject { Spree::OrderCancellations.new(order).cancel_unit(inventory_unit) }
+    let(:order) { create(:shipped_order, line_items_count: 1) }
+    let(:inventory_unit) { order.inventory_units.first }
+
+    it "creates a UnitCancel record" do
+      expect { subject }.to change { Spree::UnitCancel.count }.by(1)
+
+      unit_cancel = Spree::UnitCancel.last
+      expect(unit_cancel.inventory_unit).to eq inventory_unit
+      expect(unit_cancel.reason).to eq Spree::UnitCancel::DEFAULT_REASON
+    end
+
+    it "cancels the inventory unit" do
+      expect { subject }.to change { inventory_unit.state }.to "canceled"
+    end
+
+    context "with a reason" do
+      subject { order.cancellations.cancel_unit(inventory_unit, reason: "some reason") }
+
+      it "sets the reason on the UnitCancel" do
+        expect { subject }.to change { Spree::UnitCancel.count }.by(1)
+        expect(Spree::UnitCancel.last.reason).to eq("some reason")
```

Comment:
Can we base this on the return from subject rather than `Spree::UnitCancel.last`.